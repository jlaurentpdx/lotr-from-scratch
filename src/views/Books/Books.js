import { useState, useEffect } from 'react';
import BookList from '../../components/BookList';
import { fetchBooks } from '../../services/books';
import './Books.css';

export default function Books() {
  const [books, setBooks] = useState([]);

  useEffect(() => {
    const fetchData = async () => {
      const data = await fetchBooks();
      setBooks(data);
    };
    fetchData();
  }, []);

  // console.log(books);

  return <div>Books page</div>;
}
