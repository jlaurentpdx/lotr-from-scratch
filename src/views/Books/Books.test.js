import { render, screen } from '@testing-library/react';
import Books from './Books';

test('Books should render three titles', async () => {
  const { container } = render(<Books />);
  await screen.findByText('The Fellowship of the Ring');
  await screen.findByText('The Two Towers');
  await screen.findByText('The Return of the King');

  expect(container).toMatchSnapshot();
});
