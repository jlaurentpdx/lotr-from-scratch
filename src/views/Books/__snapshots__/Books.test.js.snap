// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Books should render three titles 1`] = `
<div>
  <div
    class="books-page"
  >
    <h2>
      Books
    </h2>
    <div
      class="book-list"
    >
      <h3>
        The Fellowship of the Ring
      </h3>
      <h3>
        The Two Towers
      </h3>
      <h3>
        The Return of the King
      </h3>
    </div>
  </div>
</div>
`;
